projects/can_transceiver/src/can_transceiver_ros_intf.cpp

```diff
@@ -43,13 +69,131 @@ class CanTransceiverIntf : public rclcpp::Node
 
     /**
      * @brief Callback function to subscribe to the onboard ROS network
-     * 
+     *
      */
     void sub_cb(std_msgs::msg::String /* placeholder */) { can_trns_->onNewCmd(Placeholder0); }
 };
 
+
+//===========================================================================================
+// Simulation Interface
+//===========================================================================================
+/* Refer to https://ubcsailbot.atlassian.net/wiki/spaces/prjt22/pages/1768849494/Simulation+Interface#Interfaces
+ * Handles publishing and subscribing to certain ROS topics
+ */
+class CanSimIntf : public rclcpp::Node
+{
+public:
+    CanSimIntf()  //Our node which subs to topics
+    : Node("CanSimIntf")
+    {
+    // Subscriber
+    // There is no timer because subscriber will respond to whatever data is published to the topic /Simulator
+        // Topic: mock_gps
+        subscriptionGPS_ = this->create_subscription<custom_interfaces::msg::GPS>(
+          "mock_gps", QUEUE_SIZE, std::bind(&CanSimIntf::gps_callback, this, std::placeholders::_1));
+
+        // Topic: mock_wind_sensors
+        subscriptionWindSensor_ = this->create_subscription<custom_interfaces::msg::WindSensors>(
+          "mock_wind_sensors", QUEUE_SIZE, std::bind(&CanSimIntf::wind_callback, this, std::placeholders::_1));
+
+    // Publisher
+        publisherWindSensors_ = this->create_publisher<custom_interfaces::msg::WindSensors>("wind_sensors", QUEUE_SIZE);
+        publisherWindSensor_ = this->create_publisher<custom_interfaces::msg::WindSensor>("filtered_wind_sensor", QUEUE_SIZE);
+        publisherGPS_ = this->create_publisher<custom_interfaces::msg::GPS>("gps", QUEUE_SIZE);
```

Comment:
Use the constants defined under lib/cmn_hdrs/ros_info.h

Reply:
Replaced